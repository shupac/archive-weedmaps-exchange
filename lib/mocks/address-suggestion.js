export const mockSuggestedAddresses = [
  {address: "41 Discovery Park Boulevard, Seattle, WA, USA", id: "Ei00MSBEaXNjb3ZlcnkgUGFyayBCb3VsZXZhcmQsIFNlYXR0bGUsIFdBLCBVU0E"},
  {address: "41 Discovery Drive, The Dalles, OR, USA", id: "Eic0MSBEaXNjb3ZlcnkgRHJpdmUsIFRoZSBEYWxsZXMsIE9SLCBVU0E"},
  {address: "41 Discovery Circle Northeast, Issaquah, WA, USA", id: "EjA0MSBEaXNjb3ZlcnkgQ2lyY2xlIE5vcnRoZWFzdCwgSXNzYXF1YWgsIFdBLCBVU0E"},
  {address: "41 Discovery Trail, Vancouver, WA, USA", id: "EiY0MSBEaXNjb3ZlcnkgVHJhaWwsIFZhbmNvdXZlciwgV0EsIFVTQQ"},
  {address: "41 Discovery Street, Walla Walla, WA, USA", id: "Eik0MSBEaXNjb3ZlcnkgU3RyZWV0LCBXYWxsYSBXYWxsYSwgV0EsIFVTQQ"},
]
