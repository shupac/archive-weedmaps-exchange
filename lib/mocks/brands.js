export const mockBrands = [
  {
    slug: 'gfarmalabs',
    name: 'GfarmaLabs',
    description:
      '<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> <p> </p> <p>G FarmaLabs has become a well known brand throughout the cannabis community with both cannabis patients and connoisseurs alike. G FarmaLabs is committed to applying science and global resources to design and develop market leading brands in the cannabis flower, extracts, and edible product areas.</p> <p><strong><ins>History:</ins></strong></p> <p>Family owned and operated, G Farmalabs was conceived in 2007 by husband and wife team Ata and Nicole Gonzalez. Forced to relocate to California in 2009 after the financial crisis known as “The Great Recession,” Ata and Nicole believed the end of marijuana prohibition was near. Upon arriving in California, Ata attended Oaksterdam University and rented their first farm in Northern California. Together they carefully planted and harvested their first batch of marijuana flowers. Nicole is a member of a fourth generation tobacco family and has a strong background in the cigar industry. In 2013, G FarmaLabs launched America’s first complete brand of cannabis products.</p> <p>In 2015, Ata and Nicole’s idea had now become reality. G FarmaBrands was born. G FarmaLabs is the world’s first cannabis company that dedicates their effort to following the traditional lessons adapted from the Tobacco Industry. G FarmaLabs infused the two commercial enterprises to create a variety of products in an emerging market, setting the trend for the future of cannabis by merging the two industries.</p> <p><strong><ins>Process:</ins></strong></p> <p>G FarmaLabs works closely with a team of professionals who designed the California based processing center and production facility called the “Innovation Center” located in Desert Hot Springs, California. They have adopted the phrase “Napa Valley of Cannabis” and focus on blending the finest Domestic Cultivated Cannabis. Consistently cultivated, G FarmaLabs uses the same secret strains, each carefully chosen to guarantee a consistent experience.</p> <p>Perfecting the recipe and process, the G FarmaLabs farmers cultivate, harvest, dry, and trim their seven strains making it possible for customers to experience the aroma and feel the passion and precision the team at G FarmaLabs dedicates to their line of products.</p> <p><strong><ins>Products:</ins></strong></p> <p>Offering a variety of products from pre-rolls to edibles, G FarmaLabs caters to a wide market of cannabis users. G Stiks, their line of prestigious, 100% indoor grown pre-rolled joints are grown, processed, and rolled by G FarmaLabs and are available in 6 different options depending on user preference. For people who prefer Indica, G FarmaLabs G Stiks offers “ROJO”, packaged in red symbolizing a potent form of medicine. Patients in medical marijuana states enjoy the highly desired “VIOLETA” G Stik, packaged in purple, similar to violet, symbolizing various things from royalty to wealth, magic and mystery. The “AMARILLO” Sativa G Stiks symbolize a yellow butterfly purposely chosen to create a euphoric feeling for patients who suffer from depression. The “ROSADO” Sativa G Stik symbolizing one of the most celebrated flowers, the Pink Lotus Flower and is designed to uplift the heart and mind. The “VERDE” G Stik symbolizes energy and healing and the “NARAN-JA” G Stik symbolizes the Orange Rose. G FarmaLabs’ hybrid G Stiks offer a solution for connoisseurs on the go who need a product that is ready to go. The “VERDE” blend promotes energy healing and stress relief.</p> <p>In addition to G Stiks, G FarmaLabs also offers a line of vapor products, injectors, and edibles. Their Liquid Gold vape battery pens support 1.5 grams and are available in Super Silver Haze, Silver Haze, Blue Diamond, Green Crack , Orange Kush, Red Diesel, Pink Lemonade, Blackberry Kush, White Widow, and Purple Haze. They also proudly offer Indica, Sativa, and Hybrid Red Cannabis Oil Injectors in 0.5-1 gram dosages as well as NUG RUN Cannabis Oil injectors that are currently only offered in a 0.5 gram dosages designed to allow users to inhale Cannabinoids while avoiding harmful smoke toxins. G FarmaLabs offers infused decadent edible chocolate bars and truffles and 100% raw and unfiltered honey.</p> <p>From the detail of their packaging to their meticulous formula, G FarmaLabs proudly manufactures their own products delivering only the highest quality cannabis.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Recognized as a Leader in the American Cannabis Revolution, G FarmaLabs has influenced celebrities such as The Game and has been featured on ABC News 7 and 790 KABC Talk Radio. G FarmaLabs can be found in dispensaries throughout Northern California, San Bernardino, Los Angeles, Orange County, Southern California, and Washington.</p>',
    id: 'd054b34f-428e-4583-a2a4-27f4e5aea6f3',
  },
  {
    slug: 'thclear-co',
    name: 'THClear Co',
    description:
      '<p>We are the industry leader in solventless distillate. We currently produce 6 different products with our Thclear Distillate, with our infamous uncut Private Reserve Cartridge being the most popular. Our other amazing products include: Disposable Pens, Cartridges, Syringes, Honeypot Vials, and our funky 50/50!</p>',
    id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
  },
];
