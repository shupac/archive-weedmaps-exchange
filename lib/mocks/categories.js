
export const mockCategories = [
  {
    "slug":"concentrates",
    "position":1,
    "name":"Concentrates",
    "id":"81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "id":"c77396fe-4293-4911-805c-2cd3afb18949"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "id":"82c8a744-10c6-48c9-b5a0-3a90de6d2c2f"
    }
  },
  {
    "slug":"edibles",
    "position":3,
    "name":"Edibles",
    "id":"e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "id":"b252a9f9-f06e-4981-80db-cf74d6c7573a"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "id":"97cc29eb-0ce4-4b55-81ae-e1899b6dc7ce"
    }
  }
];