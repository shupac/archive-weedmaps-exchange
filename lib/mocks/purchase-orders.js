const mockPurchaseOrders = [
  {
    id: "3b4b09e2-4f0f-4c83-8b1c-2760f3af12f9",
    total: "25000",
    subtotal: "24900.00",
    statusReason: "",
    status: "not_started",
    shippingFee: "100.00",
    sellerData: {
      sellerPhone: "1112223333",
      sellerName: "GfarmaLabs",
      sellerLicenses: [],
      sellerId: "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
      sellerEmail: "test@test.com",
      sellerAddress: "{\"territory\":\"OR\",\"street_address\":\"120 Columbia Ave NE\",\"region\":{\"name\":\"Washington\",\"id\":\"bd457c68-fee7-485b-b3ab-c2daabba3e4f\"},\"postal_code\":\"97818\",\"country\":\"us\",\"city\":\"Boardman\"}",
      brandId: "d054b34f-428e-4583-a2a4-27f4e5aea6f3"
    },
    orderDate: "2018-11-27T21:52:52.332967Z",
    givenDeliveryEta: {
      etaMinUnit: "day",
      etaMin: 1,
      etaMaxUnit: "day",
      etaMax: 3
    },
    expectedShipDateMin: "2018-11-28T21:52:52.321156Z",
    expectedShipDateMax: "2018-11-30T21:52:52.321156Z",
    buyerData: {
      buyerPhone: "5555555555",
      buyerLocationName: "Sciametta Burweedos",
      buyerLicenses: [
        {
          number: "888888888",
          licenseType: "medical",
          id: "6099e0b7-c7f9-4894-8df6-639469df8e5d"
        }
      ],
      buyerEmail: "joesmith@aol.com",
      buyerDeliveryInstructions: "Delivery Instruction",
      buyerAddress: "{\"territory\":\"CA\",\"street_address\":\"41 Discovery\",\"region\":{\"name\":\"South OC Brands\",\"id\":\"64d05017-4339-4cda-9e57-0da061bf6b00\"},\"postal_code\":\"92618\",\"country\":\"us\",\"city\":\"Irvine\"}"
    },
    orderItems: [
      {
        id: "eea69c02-9a58-40d5-9ea5-275c4059055e"
      }
    ]
  },
  {
    id: "11690783-9715-4ec7-a184-f0b25638fb05",
    total: "115000",
    subtotal: "114900.00",
    statusReason: "Not needed",
    status: "canceled",
    shippingFee: "100.00",
    sellerData: {
      sellerPhone: "1112223333",
      sellerName: "THClear Co",
      sellerLicenses: [],
      sellerId: "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
      sellerEmail: "test@test.com",
      sellerAddress: "{\"territory\":\"OR\",\"street_address\":\"120 Columbia Ave NE\",\"region\":{\"name\":\"Washington\",\"id\":\"bd457c68-fee7-485b-b3ab-c2daabba3e4f\"},\"postal_code\":\"97818\",\"country\":\"us\",\"city\":\"Boardman\"}",
      brandId: "d60ca06a-89db-4584-99a8-1f9b9f427e42"
    },
    orderDate: "2018-11-27T20:27:02.311382Z",
    givenDeliveryEta: {
      etaMinUnit: "day",
      etaMin: 1,
      etaMaxUnit: "day",
      etaMax: 3
    },
    expectedShipDateMin: "2018-11-28T20:27:02.297357Z",
    expectedShipDateMax: "2018-11-30T20:27:02.297357Z",
    buyerData: {
      buyerPhone: "5555555555",
      buyerLocationName: "Sciametta Burweedos",
      buyerLicenses: [
        {
          number: "888888888",
          licenseType: "medical",
          id: "6099e0b7-c7f9-4894-8df6-639469df8e5d"
        }
      ],
      buyerEmail: "joesmith@aol.com",
      buyerDeliveryInstructions: "Delivery Instruction",
      buyerAddress: "{\"territory\":\"CA\",\"street_address\":\"41 Discovery\",\"region\":{\"name\":\"South OC Brands\",\"id\":\"64d05017-4339-4cda-9e57-0da061bf6b00\"},\"postal_code\":\"92618\",\"country\":\"us\",\"city\":\"Irvine\"}"
    },
    orderItems: [
      {
        id: "87623787-bdbe-4ec1-8dba-b39e99d4d597"
      }
    ]
  },

]

export default mockPurchaseOrders