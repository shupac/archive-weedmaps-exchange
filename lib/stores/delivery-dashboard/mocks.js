export const mockIndex = {
  data: {
    id: '1',
    type: 'index_card',
    relationships: {
      driver_cards: {
        data: [
          {
            type: 'driver_card',
            id: '1',
          },
          {
            type: 'driver_card',
            id: '2',
          },
          {
            type: 'driver_card',
            id: '3',
          },
        ],
      },
      route_cards: {
        data: [
          {
            type: 'route_card',
            id: '1',
          },
        ],
      },
      delivery_cards: {
        data: [
          {
            type: 'delivery_card',
            id: '1',
          },
          {
            type: 'delivery_card',
            id: '2',
          },
        ],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Alan Tirado',
        contact_email: 'atirado@weedmaps.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '123-123-1234',
        item_count: 4,
        order_id: '1234',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'created',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '2',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '3',
      relationships: {
        package: {
          data: {
            id: '3',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'assigned',
      },
      id: '1',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '1',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '1',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'in_progress',
      },
      id: '2',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '3',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '2',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [
            {
              id: '1',
              type: 'route_card',
            },
          ],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'James',
        phone_number: '123-123-1234',
        avatar_url:
          'https://images.weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg?development=1',
      },
      id: '2',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'off_duty',
        name: 'Val',
        phone_number: '666-666-5656',
        avatar_url: null,
      },
      id: '3',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockBaseIndex = {
  data: [
    {
      id: '1',
      type: 'index_card',
      relationships: {
        driver_cards: {
          data: [
            {
              type: 'driver_card',
              id: '1',
            },
          ],
        },
        route_cards: {
          data: [],
        },
        delivery_cards: {
          data: [
            {
              type: 'delivery_card',
              id: '1',
            },
          ],
        },
      },
    },
  ],
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'successful',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'String',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: 'String',
        item_count: 4,
        order_id: 'String',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'free',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockNewRoute = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'assigned',
    },
    id: '1',
    relationships: {
      delivery_cards: {
        data: [
          {
            id: '1',
            type: 'delivery_card',
          },
        ],
      },
      driver: {
        data: {
          id: '1',
          type: 'driver_card',
        },
      },
    },
  },
  included: [],
  jsonapi: { version: '1.0' },
};
export const mockNewDelivery = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'created',
      destination_address: '41 Discovery',
      destination_latitude: 33.666629,
      destination_longitude: -117.756304,
      recipient_name: 'Alan Tirado',
      ordered_at: '2018-05-08T23:00:42.075253Z',
      contact_number: '123-123-1234',
      item_count: 4,
      order_id: '420',
      total: 420,
    },
    id: '2',
    relationships: {
      package: {
        data: {
          id: '2',
          type: 'package',
        },
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockNewDriver = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'off_duty',
      name: 'Bob',
      phone_number: '123-123-1234',
      avatar_url:
        'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
    },
    id: '3',
    type: 'driver_card',
    relationships: {
      in_progress_route_cards: {
        data: [],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockDriverLocation = {
  data: {
    id: '1',
    type: 'driver_location',
    attributes: {
      latitude: 33.666629,
      longitude: -117.756304,
      occurred_at: '2018-05-08T23:00:42.075253Z',
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockPackage = {
  data: {
    attributes: {
      contact_email: 'labron@weedmaps.com',
      contact_number: '555-555-5555',
      documents: [
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg',
          name: 'id',
        },
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/002/306/378/avatar/square_fill/1531352592-download.jpeg',
          name: 'rec',
        },
      ],
      excise_tax: 1,
      inserted_at: '2018-05-08T23:00:42.075253Z',
      order_id: 'UUID-1234',
      order_items: [
        {
          name: 'Hardcore Og',
          price: 50,
          quantity: 1,
        },
      ],
      ordered_at: '2018-05-08T23:00:42.075253Z',
      recipient_name: 'Bob',
      recipient_note: 'Quick yo!',
      sales_tax: 1,
      subtotal: 1,
      total: 1,
      updated_at: '2018-05-08T23:00:42.075253Z',
    },
    id: '1234',
    type: 'package',
  },
  jsonapi: {
    version: '1.0',
  },
};
