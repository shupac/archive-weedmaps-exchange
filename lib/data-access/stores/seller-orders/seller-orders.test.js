import * as MST from 'mobx-state-tree';
import mockPurchaseOrders from 'lib/mocks/purchase-orders';
import SellerOrders from './';
import { ordersBrands, ordersLocations } from '../../../mocks/purchase-orders';

const { getSnapshot } = MST;

function setup(mockResp) {
  const mockFetchClient = {
    fetch: jest.fn().mockReturnValue(Promise.resolve({ data: mockResp })),
    put: jest.fn().mockReturnValue(Promise.resolve({ data: mockResp })),
    post: jest.fn().mockReturnValue(Promise.resolve({ data: mockResp })),
  };

  const mockStore = {
    sellerOrders: SellerOrders.create(
      {
        ordersList: mockPurchaseOrders,
      },
      {
        client: mockFetchClient,
      },
    ),
  };

  return { mockStore, mockFetchClient };
}

describe('Seller Orders Store', () => {
  it('can fetch all purchase orders and set data', async () => {
    const { mockStore } = setup(mockPurchaseOrders);
    const { sellerOrders } = mockStore;
    await sellerOrders.fetchOrders();
    expect(getSnapshot(sellerOrders.ordersList)).toMatchSnapshot();
  });

  it('can compute sellerBrandsFilterOptions', async () => {
    const { mockStore } = setup(ordersBrands);
    const { sellerOrders } = mockStore;
    await sellerOrders.fetchPOBuyers();
    expect(sellerOrders.sellerBrandsFilterOptions).toEqual([
      {
        value: 'd054b34f-428e-4583-a2a4-27f4e5aea6f3',
        text: 'GfarmaLabs',
      },
    ]);
  });

  it('can compute sellerLocationsFilterOptions', async () => {
    const { mockStore } = setup(ordersLocations);
    const { sellerOrders } = mockStore;
    await sellerOrders.fetchPOBuyers();
    expect(sellerOrders.sellerLocationsFilterOptions).toEqual([
      {
        value: '244a4623-009f-48de-b294-9d463b9973c6',
        text: 'SF Location',
      },
    ]);
  });

  it('can compute sellerZonesFilterOptions', async () => {
    const { mockStore } = setup(ordersLocations);
    const { sellerOrders } = mockStore;
    await sellerOrders.fetchPOBuyers();
    expect(sellerOrders.sellerZonesFilterOptions).toEqual([
      {
        value: '244a4623-009f-48de-b294-9d463b9973c6',
        text: 'SF Location',
      },
    ]);
  });
});
