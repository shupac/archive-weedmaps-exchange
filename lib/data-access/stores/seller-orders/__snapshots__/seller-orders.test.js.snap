// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seller Orders Store can fetch all purchase orders and set data 1`] = `
Array [
  Object {
    "buyerData": Object {
      "buyerAddress": "{\\"territory\\":\\"CA\\",\\"street_address\\":\\"41 Discovery\\",\\"region\\":{\\"name\\":\\"South OC Brands\\",\\"id\\":\\"64d05017-4339-4cda-9e57-0da061bf6b00\\"},\\"postal_code\\":\\"92618\\",\\"country\\":\\"us\\",\\"city\\":\\"Irvine\\"}",
      "buyerDeliveryInstructions": "Delivery Instruction",
      "buyerEmail": "joesmith@aol.com",
      "buyerLicenses": Array [
        Object {
          "id": "6099e0b7-c7f9-4894-8df6-639469df8e5d",
          "licenseType": "medical",
          "number": "888888888",
        },
      ],
      "buyerLocationName": "Sciametta Burweedos",
      "buyerName": "Andrew",
      "buyerPhone": "5555555555",
    },
    "expectedShipDateMax": "2018-11-30T21:52:52.321156Z",
    "expectedShipDateMin": "2018-11-28T21:52:52.321156Z",
    "givenDeliveryEta": Object {
      "etaMax": 3,
      "etaMaxUnit": "day",
      "etaMin": 1,
      "etaMinUnit": "day",
    },
    "id": "3b4b09e2-4f0f-4c83-8b1c-2760f3af12f9",
    "orderDate": "2018-11-27T21:52:52.332967Z",
    "orderItems": Array [
      Object {
        "id": "eea69c02-9a58-40d5-9ea5-275c4059055e",
      },
    ],
    "sellerData": Object {
      "sellerAddress": "{\\"territory\\":\\"OR\\",\\"street_address\\":\\"120 Columbia Ave NE\\",\\"region\\":{\\"name\\":\\"Washington\\",\\"id\\":\\"bd457c68-fee7-485b-b3ab-c2daabba3e4f\\"},\\"postal_code\\":\\"97818\\",\\"country\\":\\"us\\",\\"city\\":\\"Boardman\\"}",
      "sellerEmail": "test@test.com",
      "sellerId": "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
      "sellerLicenses": Array [],
      "sellerName": "GfarmaLabs",
      "sellerPhone": "1112223333",
    },
    "shippingFee": "100.00",
    "status": "not_started",
    "statusReason": "",
    "subtotal": "24900.00",
    "total": "25000",
    "zoneName": "Zone1",
  },
  Object {
    "buyerData": Object {
      "buyerAddress": "{\\"territory\\":\\"CA\\",\\"street_address\\":\\"41 Discovery\\",\\"region\\":{\\"name\\":\\"South OC Brands\\",\\"id\\":\\"64d05017-4339-4cda-9e57-0da061bf6b00\\"},\\"postal_code\\":\\"92618\\",\\"country\\":\\"us\\",\\"city\\":\\"Irvine\\"}",
      "buyerDeliveryInstructions": "Delivery Instruction",
      "buyerEmail": "joesmith@aol.com",
      "buyerLicenses": Array [
        Object {
          "id": "6099e0b7-c7f9-4894-8df6-639469df8e5d",
          "licenseType": "medical",
          "number": "888888888",
        },
      ],
      "buyerLocationName": "Sciametta Burweedos",
      "buyerName": "Andrew",
      "buyerPhone": "5555555555",
    },
    "expectedShipDateMax": "2018-11-30T20:27:02.297357Z",
    "expectedShipDateMin": "2018-11-28T20:27:02.297357Z",
    "givenDeliveryEta": Object {
      "etaMax": 3,
      "etaMaxUnit": "day",
      "etaMin": 1,
      "etaMinUnit": "day",
    },
    "id": "11690783-9715-4ec7-a184-f0b25638fb05",
    "orderDate": "2018-11-27T20:27:02.311382Z",
    "orderItems": Array [
      Object {
        "id": "87623787-bdbe-4ec1-8dba-b39e99d4d597",
      },
    ],
    "sellerData": Object {
      "sellerAddress": "{\\"territory\\":\\"OR\\",\\"street_address\\":\\"120 Columbia Ave NE\\",\\"region\\":{\\"name\\":\\"Washington\\",\\"id\\":\\"bd457c68-fee7-485b-b3ab-c2daabba3e4f\\"},\\"postal_code\\":\\"97818\\",\\"country\\":\\"us\\",\\"city\\":\\"Boardman\\"}",
      "sellerEmail": "test@test.com",
      "sellerId": "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
      "sellerLicenses": Array [],
      "sellerName": "THClear Co",
      "sellerPhone": "1112223333",
    },
    "shippingFee": "100.00",
    "status": "canceled",
    "statusReason": "Not needed",
    "subtotal": "114900.00",
    "total": "115000",
    "zoneName": "Zone1",
  },
]
`;

exports[`Seller Orders Store can fetch purchase order and set data 1`] = `
Object {
  "buyerData": Object {
    "buyerAddress": "{\\"territory\\":\\"CA\\",\\"street_address\\":\\"55 Discovery\\",\\"region\\":{\\"name\\":\\"South OC Brands\\",\\"id\\":\\"64d05017-4339-4cda-9e57-0da061bf6b00\\"},\\"postal_code\\":\\"92618\\",\\"country\\":\\"us\\",\\"city\\":\\"Irvine\\"}",
    "buyerDeliveryInstructions": "TEST.",
    "buyerEmail": "asciametta@aol.com",
    "buyerLicenses": Array [
      Object {
        "id": "e5ae232c-9ae7-4e81-b6f6-67bd5562706f",
        "licenseType": "medical",
        "number": "THX1138",
      },
    ],
    "buyerLocationName": "Andrew's Cryb",
    "buyerName": "Andrew",
    "buyerPhone": "(303) 555-5555",
  },
  "expectedShipDateMax": "2018-11-15T17:48:29.169876Z",
  "expectedShipDateMin": "2018-11-13T17:48:29.169876Z",
  "givenDeliveryEta": Object {
    "etaMax": 6,
    "etaMaxUnit": "day",
    "etaMin": 4,
    "etaMinUnit": "day",
  },
  "id": "b97329d4-a7ae-4c7a-ab5e-4de8aec22f50",
  "orderDate": "2018-11-09T17:48:29.174071Z",
  "orderItems": Array [
    Object {
      "id": "2b113f61-1b7f-4d18-bd20-7c3ebc7d2007",
    },
    Object {
      "id": "7ada039c-fca7-462c-b87c-6d69a815d442",
    },
  ],
  "sellerData": Object {
    "sellerAddress": null,
    "sellerEmail": "test@test.com",
    "sellerId": "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
    "sellerLicenses": Array [],
    "sellerName": "West Coast Cure",
    "sellerPhone": "1112223333",
  },
  "shippingFee": "0.0",
  "status": "not_started",
  "statusReason": "",
  "subtotal": "4400.0",
  "total": "4400",
  "zoneName": "Zone1",
}
`;
